pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/MinAggregationFunction.java

```diff
@@ -48,72 +47,158 @@ public void accept(@Nonnull AggregationFunctionVisitorBase visitor) {
   @Nonnull
   @Override
   public AggregationResultHolder createAggregationResultHolder() {
-    return new DoubleAggregationResultHolder(DEFAULT_VALUE);
+    return new ObjectAggregationResultHolder();
   }
 
   @Nonnull
   @Override
   public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity, int trimSize) {
-    return new DoubleGroupByResultHolder(initialCapacity, maxCapacity, trimSize, DEFAULT_VALUE, true);
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity, trimSize);
```

Comment:
Should we have a comparable result holder?
Also, for group-by we need to measure any performance issues due to moving from primitive double[] to Object[].